src/packaging/bin/spreaper

```diff
@@ -397,12 +411,50 @@ class ReaperCLI(object):
             print("# HTTP request failed with err: {}".format(err))
             exit(2)
 
+    @staticmethod
+    def prepare_reaper_for_login(command, description, usage=None, extra_arguments=None):
+        (args, ignored) = _parse_arguments(command, description, usage, extra_arguments)
+        reaper = ReaperCaller(args.reaper_host, args.reaper_port, args.reaper_use_ssl)
+        return reaper, args
+
     @staticmethod
     def prepare_reaper(command, description, usage=None, extra_arguments=None):
         args = _parse_arguments(command, description, usage, extra_arguments)
         reaper = ReaperCaller(args.reaper_host, args.reaper_port, args.reaper_use_ssl)
+        ReaperCLI.login(reaper, args)
         return reaper, args
 
+    @staticmethod
+    def login(reaper, args):
+        if (args.username != None):
+            reaper, args = ReaperCLI.prepare_reaper_for_login(
+                "login",
+                "Authenticate to Reaper"
+            )
+            password = ReaperCLI.get_password()
+            printq("# Logging in...")
+            payload = {'username':args.username, 'password':password, 'rememberMe':False}
+            reply = reaper.postFormData("login", 
+                                payload=payload)
+            # Use shiro's session id to request a JWT
+            COOKIES["JSESSIONID"] = reply.cookies["JSESSIONID"]
+            jwt = reaper.get("jwt")
```

Comment:
We want a way to read the `REAPER_JWT_TOKEN` from either a environment variable or the `~/.reaper` file, so to avoid the login request (which sends the password over the wire every spreaper call).

Reply:
Setting an env variable that can be reused between calls doesn't seem to be possible.
I've just pushed a change to store the JWT in `~/.reaper_jwt` on top of accepting the `--jwt` arg.
The token isn't displayed anymore and the user is just told he is now authenticated to Reaper. Any 403 error would display a message asking to login again.

Reply:
Thanks.

Wasn't thinking about setting the env variable here, just being able to read one already set by the user.

Will test it over the weekend. Am I free to squash, rebase and merge?

Reply:
Sure thing